core/server/web/api/v2/content/app.js

```diff
@@ -0,0 +1,62 @@
+// # API routes
+const debug = require('ghost-ignition').debug('api'),
+    boolParser = require('express-query-boolean'),
+    express = require('express'),
+
+    // routes
+    routes = require('./routes'),
+
+    // Include the middleware
+
+    // API specific
+    versionMatch = require('../../../middleware/api/version-match'), // global
+
+    // Shared
+    bodyParser = require('body-parser'), // global, shared
+    cacheControl = require('../../../middleware/cache-control'), // global, shared
+    maintenance = require('../../../middleware/maintenance'), // global, shared
+    errorHandler = require('../../../middleware/error-handler'); // global, shared
+
+module.exports = function setupApiApp() {
+    debug('Content API v2 setup start');
+    const apiApp = express();
+
+    // @TODO finish refactoring this away.
+    // NOTE: does this apply for both public and admin APIs?
```

Comment:
@kirrg001 would love a comment on this question :smiley:

Reply:
This is used in two places currently. And we can get rid of it.

One place is the private blogging app, which can be removed. This is an old leftover, because the see this [commit](https://github.com/TryGhost/Ghost/commit/c1a260151482de47745f424f150543a7a9e47913) - the private blogging app was once part of one express registration, but now it's only registered for the site express app. Private blogging is a proper app now. So the usage can be removed. I will push the removal straight to master.

The other place is the [redirect middleware](https://github.com/TryGhost/Ghost/blob/2.1.2/core/server/web/middleware/url-redirects.js), which decides whether to redirect the request or not e.g. you have configured a specific admin host, but you access the url of the admin with a wrong host. Same for http vs https.
We can get rid of res.isAdmin completely, if we would register different middlewares for different express apps. Blog App -> getBlogRedirectUrl and Admin App -> getAdminRedirectUrl. This applies only for v2, v0.1 stays as is.

If you rebase against master, you can remove isAdmin from v2 and register the different middlewares for admin & content. We probably need a little wrapper around each middleware, because the wrapper needs to determine if we need a redirect, we don't want to modify the middleware implementation. Similar to the wrapper which exists - which handles blog or admin requests.

Reply:
This comment is outstanding. @gargol

Reply:
Added this improvement as a task on our gigantic :chart_with_upwards_trend:  TODO list